components/ListView/SourcesListItem.js

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { FormattedMessage, defineMessages, injectIntl, intlShape } from "react-intl";
+
+const messages = defineMessages({
+  edit: {
+    defaultMessage: "Edit source"
+  },
+  remove: {
+    defaultMessage: "Remove source"
+  },
+  baseurl: {
+    defaultMessage: "yum repository"
+  },
+  mirrorlist: {
+    defaultMessage: "mirrorlist"
+  },
+  metalink: {
+    defaultMessage: "metalink"
+  }
+});
+
+class SourcesListItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.editButton = React.createRef();
+  }
+  componentDidMount() {
+    if (this.props.edited === this.props.source.name) {
+      this.editButton.current.focus();
+    }
+  }
+  render() {
+    const { source, key } = this.props;
+    const type = source.type.substring("yum-".length);
+    const { formatMessage } = this.props.intl;
+    return (
+      <div className="list-pf-item" key={key}>
```

Comment:
I'm not sure. I customized the layout, and am not sure how easy it would be to include that with the PF3 react component. With PF4 being available and with differences between the list views in PF3 and PF4, I'm thinking I can refactor all the list views at the same time when migrating to PF4.